lib/bitcoin/TransactionNumber.ts

```diff
@@ -1,42 +1,68 @@
+import ErrorCode from './ErrorCode';
+import SidetreeError from '../common/SidetreeError';
+
 /**
- * Defines the TransactionNumber as a combination of blockNumber and position within the block
+ * Defines the TransactionNumber as a combination of block number and transaction index within the block.
  */
 export default class TransactionNumber {
 
-  /* We set blockNumber and position to be 32 bits each */
-  private static readonly bitWidth = 32;
+  /**
+   * Maximum allowed transaction index in a block.
+   */
+  private static readonly maxTransactionIndexInBlock = 999999;
 
   /**
-   * Constructs the transaction number given the block number and position of the transaction in the block.
+   * Maximum allowed transaction count in a block.
    */
-  public static construct (blockNumber: number, position: number) {
-    const transactionNumber =
-      blockNumber * (2 ** this.bitWidth) +
-      position;
+  private static readonly maxTransactionCountInBlock = TransactionNumber.maxTransactionIndexInBlock + 1; // 1,000,000
 
+  /**
+   * Constructs the transaction number given the block number and the transaction index in the block.
+   */
+  public static construct (blockNumber: number, transactionIndexInBlock: number) {
+    // NOTE: JavaScript can have 53 bit integer before starting to loose precision: 2 ^ 53 = 9,007,199,254,740,992.
+    // We allocate first 6 digits for transaction index within a block, and rest of the digits for block number.
+
+    if (transactionIndexInBlock > TransactionNumber.maxTransactionIndexInBlock) {
+      throw new SidetreeError(
+        ErrorCode.TransactionNumberTransactionIndexInBlockTooLarge,
+        `Position index ${transactionIndexInBlock} given exceeds max allowed value of ${TransactionNumber.maxTransactionIndexInBlock}`
+      );
+    }
+
+    // Choosing a nice round number as long as it is less than `9007199254`.
+    const maxBlockNumber = 9000000000;
```

Comment:
Maybe something like `const JAVASCTIPT_MAX_INT = Math.pow(2, 53);` and then `const maxBlockNumber = JAVASCTIPT_MAX_INT` would describe what the comments say with code.

Reply:
Will keep it as is, because we can't assign `JAVASCTIPT_MAX_INT_PRECISION` to `maxBlockNumber`, we would need to round it down to LESS than the nearest million, what we have now is simpler of the two.